### ISSUE TITLE
AccessControlException when registering a GCS repository

### ISSUE THREAD

**Issue by dadoonet** [elastic/elasticsearch#25932]
**Elasticsearch version**: 6.0.0-beta1 (545138b)

**Plugins installed**: [ `repository-gcs` ]

**Description of the problem including expected versus actual behavior**: Unable to register a GCS repository

**Steps to reproduce**:

 1. Start elasticsearch from a GCE instance with:

```sh
bin/elasticsearch
```

 2. Register a repository:

```sh
curl -XDELETE 127.0.0.1:9200/_snapshot/gcs?pretty
curl -XPUT 127.0.0.1:9200/_snapshot/gcs?pretty -H 'Content-Type: application/json' -d '{
  "type": "gcs",
  "settings": {
    "bucket": "dadoonet",
    "service_account": "_default_"
  }
}'
```  

**Provide logs (if relevant)**:

```
[2017-07-27T10:37:31,952][WARN ][r.suppressed             ] path: /_snapshot/gcs, params: {pretty=, repository=gcs}
org.elasticsearch.repositories.RepositoryException: [gcs] failed to create repository
        at org.elasticsearch.repositories.RepositoriesService.createRepository(RepositoriesService.java:389) ~[elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at org.elasticsearch.repositories.RepositoriesService.registerRepository(RepositoriesService.java:357) ~[elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at org.elasticsearch.repositories.RepositoriesService.access$100(RepositoriesService.java:56) ~[elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at org.elasticsearch.repositories.RepositoriesService$1.execute(RepositoriesService.java:109) ~[elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at org.elasticsearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:45) ~[elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:640) ~[elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:270) ~[elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:195) [elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:130) [elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150) [elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188) [elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:569) [elasticsearch-6.0.0beta1.jar:6.0.0-beta1]
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:247) [elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:210) [elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_141]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_141]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_141]
Caused by: java.security.AccessControlException: access denied ("java.net.SocketPermission" "metadata.google.internal:80" "connect,resolve")
        at java.security.AccessControlContext.checkPermission(AccessControlContext.java:472) ~[?:1.8.0_141]
        at java.security.AccessController.checkPermission(AccessController.java:884) ~[?:1.8.0_141]
        at java.lang.SecurityManager.checkPermission(SecurityManager.java:549) ~[?:1.8.0_141]
        at java.lang.SecurityManager.checkConnect(SecurityManager.java:1051) ~[?:1.8.0_141]
        at sun.net.www.http.HttpClient.openServer(HttpClient.java:541) ~[?:?]
        at sun.net.www.http.HttpClient.<init>(HttpClient.java:242) ~[?:?]
        at sun.net.www.http.HttpClient.New(HttpClient.java:339) ~[?:?]
        at sun.net.www.http.HttpClient.New(HttpClient.java:357) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1202) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1138) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1032) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:966) ~[?:?]
        at com.google.api.client.http.javanet.NetHttpRequest.execute(NetHttpRequest.java:93) ~[?:?]
        at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:972) ~[?:?]
        at com.google.api.client.googleapis.auth.oauth2.OAuth2Utils.runningOnComputeEngine(OAuth2Utils.java:60) ~[?:?]
        at com.google.api.client.googleapis.auth.oauth2.DefaultCredentialProvider.tryGetComputeCredential(DefaultCredentialProvider.java:304) ~[?:?]
        at com.google.api.client.googleapis.auth.oauth2.DefaultCredentialProvider.getDefaultCredentialUnsynchronized(DefaultCredentialProvider.java:172) ~[?:?]
        at com.google.api.client.googleapis.auth.oauth2.DefaultCredentialProvider.getDefaultCredential(DefaultCredentialProvider.java:88) ~[?:?]
        at com.google.api.client.googleapis.auth.oauth2.GoogleCredential.getApplicationDefault(GoogleCredential.java:213) ~[?:?]
        at com.google.api.client.googleapis.auth.oauth2.GoogleCredential.getApplicationDefault(GoogleCredential.java:191) ~[?:?]
        at org.elasticsearch.repositories.gcs.GoogleCloudStorageService$InternalGoogleCloudStorageService.getDefaultCredential(GoogleCloudStorageService.java:123) ~[?:?]
        at org.elasticsearch.repositories.gcs.GoogleCloudStorageService$InternalGoogleCloudStorageService.getCredential(GoogleCloudStorageService.java:118) ~[?:?]
        at org.elasticsearch.repositories.gcs.GoogleCloudStorageService$InternalGoogleCloudStorageService.createClient(GoogleCloudStorageService.java:97) ~[?:?]
        at org.elasticsearch.repositories.gcs.GoogleCloudStorageRepository.<init>(GoogleCloudStorageRepository.java:115) ~[?:?]
        at org.elasticsearch.repositories.gcs.GoogleCloudStoragePlugin.lambda$getRepositories$1(GoogleCloudStoragePlugin.java:129) ~[?:?]
        at org.elasticsearch.repositories.RepositoriesService.createRepository(RepositoriesService.java:384) ~[elasticsearch-6.0.0-beta1.jar:6.0.0-beta1]
        ... 16 more
```

**Comment by jasontedor** [comment]
Can you look @tbrooks8?

**Comment by tbrooks8** [comment]
Should be closed by #25938. @dadoonet do you have a way of verifying?

**Comment by dadoonet** [comment]
@tbrooks8 Tested on a GCE instance and it works well. So I'm closing.
